### PR TITLE
endpointmanager: Simplify Conntrack garbage collection launch code

### PR DESCRIPTION
**Pull request by joestringer**
This code used a confusing bool and triggering global map CT inside an
endpoint loop. Shuffle the code around to make it easier to understand.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/4468)

### CODE REVIEW

**Comment by joestringer** [comment]
test-me-please

**Comment by joestringer** [comment]
test-me-please

**Comment by joestringer** [comment]
```
        ===================== ERROR - VM PROVISION FAILED =====================

        Unable to provision and start VM "runtime": exit status 1", vm, err

        =======================================================================
```

**Comment by joestringer** [comment]
test-me-please

**Comment by joestringer** [comment]
Failed to provision again. I realised it was because of go fmt.

**Comment by joestringer** [comment]
test-me-please

File: pkg/endpointmanager/conntrack.go [hunk 1]
```diff
@@ -79,51 +83,44 @@ func RunGC(e *endpoint.Endpoint, isLocal, isIPv6 bool, filter *ctmap.GCFilter) {
 	}
 }
 
+// skipGC determines whether we should skip running GC for this endpoint.
+func skipGC(e *endpoint.Endpoint) bool {
```
Comment (aanm) [inline pkg/endpointmanager/conntrack.go]: Can you change this function to be a endpoint method? (I'm only asking so we can avoid using the endpoint mutex outside of its pkg)

File: pkg/endpointmanager/conntrack.go [hunk 2]
```diff
@@ -79,51 +83,44 @@ func RunGC(e *endpoint.Endpoint, isLocal, isIPv6 bool, filter *ctmap.GCFilter) {
...
-		seenGlobal := false
 		sleepTime := time.Duration(GcInterval) * time.Second
 		for {
 			eps := GetEndpoints()
+			if len(eps) > 0 {
+				if ipv4 {
+					RunGC(nil, true, ctmap.NewGCFilterBy(ctmap.GCFilterByTime))
+				}
+				if ipv6 {
+					RunGC(nil, false, ctmap.NewGCFilterBy(ctmap.GCFilterByTime))
```
Comment (aanm) [inline pkg/endpointmanager/conntrack.go]: I think the logic of the `false` should be `true` and on line 106 it should be `false`.

**Review by aanm — DISMISSED** [review]

File: pkg/endpointmanager/conntrack.go (hunk 2 above)
Comment (joestringer) [reply pkg/endpointmanager/conntrack.go]: woops! Good catch.

**Comment by joestringer** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Review by aanm — APPROVED** [review]

**Pull request merged**